frontend/src/pages/main/home/client/components/graphic/Graphic.tsx

```diff
@@ -0,0 +1,117 @@
+import {Transaction, TransactionType} from "../../../../../../types/transactions";
+import {getTransactionForYearService} from "../../../../../../services/transaction";
+import {useAppSelector} from "../../../../../../stores/redux/hooks";
+import {selectLoggedUser} from "../../../../../../stores/redux/loggedUser/loggedUserSlice";
+import {CURRENCY_USD} from "../../../../../../common/constansts/currency";
+import {OTHER_TRANSACTIONS, THRESHOLD_UNIQUE_TRANSACTIONS} from "../../../../../../common/constansts/transactions";
+import {COLORS} from "../../../../../../common/constansts/colors";
+import Finance from "../finance/Finance";
+
+Chart.register(Tooltip, Legend, ArcElement);
+
+const initialTransactions: Transaction[] = [];
+
+const Graphic: React.FC = () => {
+
+    const loggedUser = useAppSelector(selectLoggedUser);
+    const [transactions, setTransactions] = useState<Transaction[]>(initialTransactions);
+    const [income, setIncome] = useState<number>(0);
+    const [expenses, setExpenses] = useState<number>(0);
+
+    const updateTransactions = async () => {
+        const transactionData = await getTransactionForYearService(loggedUser.id, loggedUser.userType, new Date().getFullYear(), new Date().getFullYear());
+        const updatedTransactions = mergeTransactions(transactionData);
+        setTransactions(updatedTransactions);
+        const expenses = updatedTransactions.reduce((total, transaction) => total + transaction.amount, 0);
+        setExpenses(expenses);
+        const income = transactionData.filter(transaction => transaction.TransactionType === TransactionType.TOP_UP)
+                                               .reduce((total, transaction) => total + transaction.amount, 0);
+        setIncome(income);
+    }
+
+    const mergeTransactions = (transactions: Transaction[]): Transaction[] => {
+        const mergedTransactions: { [nameTo: string]: Transaction } = {};
+        const paymentTransactions = transactions.filter(transaction => transaction.TransactionType === TransactionType.PAYMENT);
+        let numberUniqueTransactions = 0;
+        let flag = true;
+
+        paymentTransactions.forEach(transaction => {
+            const key = transaction.nameTo;
+            if (!mergedTransactions[key] && numberUniqueTransactions < THRESHOLD_UNIQUE_TRANSACTIONS) {
+                mergedTransactions[key] = {...transaction};
+                numberUniqueTransactions++;
+            } else if (!mergedTransactions[key] && flag) {
+                mergedTransactions[OTHER_TRANSACTIONS] = {...transaction, nameTo: OTHER_TRANSACTIONS};
+                flag = false;
+            } else if (!mergedTransactions[key]) {
+                mergedTransactions[OTHER_TRANSACTIONS].amount += transaction.amount;
+            } else {
+                mergedTransactions[key].amount += transaction.amount;
+            }
+        });
+        return Object.values(mergedTransactions);
+    };
+
+    const graphicData = {
```

Comment:
don't create variables in the component. Make function which triggered by useEffect when you need it. Otherwise, you recreate it every single time for each rerender